tokenserver-web/src/error.rs

```diff
@@ -0,0 +1,116 @@
+use std::fmt;
+
+use actix_web::{http::StatusCode, HttpResponse, ResponseError};
+use serde::{
+    ser::{SerializeMap, Serializer},
+    Serialize,
+};
+use syncserver_common::ReportableError;
+use thiserror::Error;
+use tokenserver_common::{ErrorLocation, TokenserverError};
+use tokenserver_db::DbError;
+
+#[derive(Debug, Error)]
+pub struct ApiError(TokenserverError);
```

Comment:
I define a top-level `ApiError` that mirrors Syncstorage's `ApiError` for consistency. I chose to do this rather than just rename `TokenserverError` to `ApiError` because `TokenserverError` has to live in `tokenserver_common`, which doesn't include Actix, and we need to define `ResponseError` for `ApiError`